sqld/src/rpc/wal_log.rs

```diff
@@ -3,18 +3,25 @@ pub mod wal_log_rpc {
     tonic::include_proto!("wal_log");
 }
 
+use std::net::SocketAddr;
 use std::sync::Arc;
 
 use crate::wal_logger::{WalLogEntry, WalLogger};
 
+use async_lock::RwLock;
 use tokio_stream::wrappers::ReceiverStream;
 use tonic::Status;
 use wal_log_rpc::wal_log_server::WalLog;
 
-use self::wal_log_rpc::{wal_log_entry::Payload, Frame, LogOffset, WalLogEntry as RpcWalLogEntry};
+use self::wal_log_rpc::{
+    wal_log_entry::Payload, Frame, HelloRequest, HelloResponse, LogOffset,
+    WalLogEntry as RpcWalLogEntry,
+};
 
 pub struct WalLogService {
     logger: Arc<WalLogger>,
+    replicas_with_hello: RwLock<Vec<SocketAddr>>,
```

Comment:
what about an `HashSet` here?

Reply:
HashSet will perform worse unless we have thousands of replicas. Both cpu wise and memory wise.

Reply:
"early optimization is the root of all evil" 😄 This is clearly not on the hot path